packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,514 @@
+     * @returns Uniforms
+     */
+    public getUniforms() {
+        return {
+            ubo: [
+                { name: "dbg_shadedDiffuseColor", size: 3, type: "vec3" },
+                { name: "dbg_shadedSpecularColorPower", size: 4, type: "vec4" }, // shadedSpecularColor, shadedSpecularPower
+                { name: "dbg_thicknessRadiusUvDistance", size: 4, type: "vec4" }, // wireframeThickness, vertexRadius, uvScale, and distanceScale
+                { name: "dbg_wireframeColor1", size: 3, type: "vec3" },
+                { name: "dbg_wireframeColor2", size: 3, type: "vec3" },
+                { name: "dbg_uvColor1", size: 3, type: "vec3" },
+                { name: "dbg_uvColor2", size: 3, type: "vec3" },
+                { name: "dbg_materialColor", size: 3, type: "vec3" },
+            ],
+            fragment: `
+                uniform vec3 dbg_shadedDiffuseColor;
+                uniform vec4 dbg_shadedSpecularColorPower;
+                uniform vec4 dbg_thicknessRadiusUvDistance;
+
+                #if DBG_MODE == 0 || DBG_MODE == 1
+                    uniform vec3 dbg_wireframeColor1;
+                #endif
+                #if DBG_MODE == 1
+                    uniform vec3 dbg_wireframeColor2;
+                #elif DBG_MODE == 2 || DBG_MODE == 3
+                    uniform vec3 dbg_uvColor1;
+                    uniform vec3 dbg_uvColor2;
+                #elif DBG_MODE == 5
+                    uniform vec3 dbg_materialColor;
+                #endif
+            `,
+        };
+    }
+
+    /**
+     * Bind the uniform buffer
+     * @param uniformBuffer Uniform buffer
+     */
+    public bindForSubMesh(uniformBuffer: UniformBuffer): void {
+        if (this._isEnabled) {
+            uniformBuffer.updateVector3("dbg_shadedDiffuseColor", this._options.shadedDiffuseColor!);
+            uniformBuffer.updateFloat4("dbg_shadedSpecularColorPower", this._options.shadedSpecularColor!.x, this._options.shadedSpecularColor!.y, this._options.shadedSpecularColor!.z, this._options.shadedSpecularPower!);
+            uniformBuffer.updateFloat4("dbg_thicknessRadiusUvDistance", this._options.wireframeThickness!, this._options.vertexRadius!, this._options.uvScale!, this._options.distanceScale!);
+            uniformBuffer.updateVector3("dbg_wireframeColor1", this._options.wireframeColor1!);
+            uniformBuffer.updateVector3("dbg_wireframeColor2", this._options.wireframeColor2!);
+            uniformBuffer.updateVector3("dbg_uvColor1", this._options.uvColor1!);
+            uniformBuffer.updateVector3("dbg_uvColor2", this._options.uvColor2!);
+            uniformBuffer.updateVector3("dbg_materialColor", this._options.materialColor!);
+        }
+    }
+
+    /**
+     * Get shader code
+     * @param shaderType "vertex" or "fragment"
+     * @returns Shader code
+     */
+    public getCustomCode(shaderType: string): Nullable<{ [pointName: string]: string }> {
+        return shaderType === "vertex"

```

Comment:
I suggest that you put all the shader literals at the top of the file as constants to avoid indentation in the strings. This will be easier to read and make the string smaller.